syne_tune/optimizer/schedulers/synchronous/hyperband_bracket.py

```diff
@@ -91,16 +77,17 @@ def assert_check_rungs(rungs: List[Tuple[int, int]]):
             [-x for x in sizes]
         ), f"Rung sizes {sizes} are not decreasing"
 
-    def is_bracket_complete(self) -> bool:
-        return self.current_rung >= self.num_rungs
-
     @property
     def num_rungs(self) -> int:
-        return len(self._rungs)
+        raise NotImplementedError
```

Comment:
not sure I understand. is this an abstract class or why don't we implement this function anymore? If yes, I'd make this class abstract as well.

Reply:
Yes, SynchronousBracket is abstract. I am avoiding using abc, abstractmethod, etc, because this has downsides (I had serialization errors), and I find abc to be overkill, it is enough to just raise NotImplementedError.